### PR TITLE
Extend test coverage of hilti::rt

### PR DESCRIPTION
**Pull request by bbannier**
This patch is part of #468 and covers `types/string.h`, `types/struct.h`, `types/time.h`, and `types/union.h`.

This series also contains a small number of bug fixes.

### CODE REVIEW

File: hilti/src/rt/tests/time.cc [hunk 1]
```diff
@@ -24,4 +25,93 @@ TEST_CASE("current_time") {
     CHECK_GE(end + 1, current_time.seconds());
 }
 
+TEST_SUITE_BEGIN("Time");
+
+TEST_CASE("bool") {
+    CHECK(Time());
```
Comment (rsmmr) [inline hilti/src/rt/tests/time.cc:31]: Shouldn't these evaluate to false?

Comment (bbannier) [reply hilti/src/rt/tests/time.cc:31]: That was my initial feeling as well, but this is exactly how this operator is designed, https://github.com/zeek/spicy/blob/0dc760db6206bcf4a41cb14a7835bd950737ca08/hilti/include/rt/types/time.h#L91-L92.

That being said, I am not sure the "unset" semantic is useful as in reality it is impossible to construct a truly unset unset `Time` value. Currently we use the valid null value of `Time` to mean "unset" (same domain). We could remove "unset" completely and leave it up to users to decide what semantics the null value should have. WDYT?

Comment (bbannier) [reply hilti/src/rt/tests/time.cc:31]: Posting that expert made me reread it and I now get your point.

We could either fix comment or impl to be consistent with the other, or remove this completely. Removal seems preferable to me.

Comment (bbannier) [reply hilti/src/rt/tests/time.cc]: Removing this operator like I did in f3ebf0 seems to work.

**Pull request merged**